src/main/java/io/github/opencubicchunks/cubicchunks/core/server/CubeProviderServer.java

```diff
@@ -108,7 +112,21 @@ public CubeProviderServer(WorldServer worldServer, ICubeGenerator cubeGen) {
         this.worldServer = worldServer;
         this.profiler = worldServer.profiler;
         try {
-            this.cubeIO = new RegionCubeIO(worldServer);
+            Path path = worldServer.getSaveHandler().getWorldDirectory().toPath();
+            if (worldServer.provider.getSaveFolder() != null) {
+                path = path.resolve(worldServer.provider.getSaveFolder());
+            }
+
+            WorldSavedCubicChunksData savedData =
+                    (WorldSavedCubicChunksData) worldServer.getPerWorldStorage().getOrLoadData(WorldSavedCubicChunksData.class, "cubicChunksData");
+
+            StorageFormatProviderBase format = StorageFormatProviderBase.REGISTRY.getValue(savedData.storageFormat);
+            if (format == null) {
+                StartupQuery.notify("unsupported storage format \"" + savedData.storageFormat + '"');
+                StartupQuery.abort();
```

Comment:
Theoretically this code could run long after world initialization, for non-overworld dimensions. This should really be done once on server start and the information should be loaded from overworld.